### PR TITLE
[WFLY-12685] Upgrade MicroProfile Health 2.1.0

### PR DESCRIPTION
**Pull request by jmesnil**
JIRA: https://issues.jboss.org/browse/WFLY-12685

### CODE REVIEW

File: microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12685[WFLY-12685]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-12734[WFLY-12734] - Upgrade smallrye-health to 2.2.0
+* https://issues.apache.org/jira/browse/EAP7-1082[EAP7-1348]
```
Comment (xstefank) [inline microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: the link seems wrong :)

Comment (jmesnil) [reply microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: oups :)

File: microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+* mailto:istraka@redhat.com[Ivan Straka]
+
+=== Testing By
+
+[X] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 2.1.0
```
Comment (istraka) [inline microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: The actual version is `2.1`

File: microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+* Support of the `mp.health.disable-default-procedures` config property is postponed to https://issues.jboss.org/browse/WFLY-12342[WFLY-12342] that will add default server procedures.
+* Porting of https://github.com/thorntail-examples/health-check/blob/f164faa787d4adb616424116a1ec84a1f28b30eb/src/main/java/io/thorntail/example/HealthChecks.java#L31[test for Thorntail health-check example] is out of scope and will be done in the context of https://issues.jboss.org/browse/WFLY-12342[WFLY-12342] that will adds a proper server probes to check the `server-state` status.
+
+== Implementation Plan
+
+== Test Plan
+
+* Run the WildFly integration basic test suite and checks there are no failures related to MicroProfle Health tests.
+* Run the `microprofile-tck/health` module from WildFly integration test suite and checks that there are no failures.
+* There are no existing tests for MicroProfile Health in https://github.com/thorntail/thorntail/tree/master/testsuite[Thorntail test suite] that needs to be ported to WildFly integration test suite.
```
Comment (istraka) [inline microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: There is [a ParserTest.java](https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-health/src/test/java/org/wildfly/swarm/health/ParserTest.java) as Rosta mentioned in [a comment](https://issues.jboss.org/browse/EAP7-1348?focusedCommentId=13807656&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13807656). There are two simple tests.

Comment (jmesnil) [reply microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: @istraka thanks for the link.

WildFly test suite and MP Health TCK already covers such basic tests in https://github.com/wildfly/wildfly/tree/master/testsuite/integration/basic/src/test/java/org/wildfly/test/integration/microprofile/health.
I don't see any benefit to add this test.

Comment (jmesnil) [reply microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: I've updated the dev analysis to mention that this test will not be added to WildFly test suite

**Comment by istraka** [comment]
@jmesnil what if there are two (liveness or readiness) checks of the same name? Right now, both checks are registered. Some warn message might be useful.  This is valid for multiple deployment / subdeployment scenario when this might happen by accident.

**Comment by jmesnil** [comment]
@istraka the name of the checks are determined when the HealthCheck probes are executed.
The health check payload would have different items with the same name but I don't think that would be a problem.

Do you have a concrete example in mind?

**Comment by istraka** [comment]
Something like `WARN ... several health checks registered under same name [name1, name2,...]`. However, I would like to leave it for another RFE since we may have a further discussion about the topic and more ideas might come up. For now, I think we are good.

File: microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+* mailto:istraka@redhat.com[Ivan Straka]
+
+=== Testing By
+
+[X] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 2.1.0
+* Upgrade smallrye-health to 2.2.0
```
Comment (istraka) [inline microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: 2.1.0

File: microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12685[WFLY-12685]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-12734[WFLY-12734] - Upgrade smallrye-health to 2.2.0
```
Comment (istraka) [inline microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: 2.1.0

File: microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+* mailto:istraka@redhat.com[Ivan Straka]
+
+=== Testing By
+
+[X] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 2.1.0
```
Comment (istraka) [inline microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: 2.1

File: microprofile/WFLY-12228_upgrade_microprofile-health_2.0.1.adoc [hunk 7]
```diff
@@ -57,6 +57,7 @@ The microprofile-health-smallrye will provide its own "reporter" that aggreated
 == Non Requirements
 
 * The integration of the health HTTP endpoints or operations in EAP or WildFly probes for Kubernetes is out of scope of this feature. It will be tracked in a separate https://issues.jboss.org/browse/CLOUD-3326[CLOUD-3326] issue.
+* Health checks in subdeployments of EAR deployment is not supported.
```
Comment (istraka) [inline microprofile/WFLY-12228_upgrade_microprofile-health_2.0.1.adoc]: Wrong file?

Comment (jmesnil) [reply microprofile/WFLY-12228_upgrade_microprofile-health_2.0.1.adoc]: yes (I moved this non-requirement to the correct file)

File: microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc (hunk 4 above)
Comment (jmesnil) [reply microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc]: fixed, thanks

**Pull request merged**